File: core/src/main/java/co/infinum/collar/Trackable.java
package co.infinum.collar;

import java.util.Map;

import co.infinum.collar.annotations.TrackAttribute;

/**
 * Invoked when {@link TrackAttribute} is set for a method parameter.
 * Given attributes will be used for the caller event
 */
public interface Trackable {

    Map<String, Object> trackableAttributes();
}

File: core/src/main/kotlin/co/infinum/collar/Trackable.kt
package co.infinum.collar

import co.infinum.collar.annotations.TrackAttribute

/**
 * Invoked when [TrackAttribute] is set for a method parameter.
 * Given attributes will be used for the caller event
 */

interface Trackable {

    fun trackableAttributes(): Map<String, Any>
}
